app/src/test/kotlin/compiler/lexer/dfa/RegexDfaTest.kt

```diff
@@ -0,0 +1,92 @@
+package compiler.lexer.dfa
+import compiler.lexer.regex.Regex
+import compiler.lexer.regex.RegexFactory
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+
+class RegexDfaTest {
+    private fun abRegexDfa(): RegexDfa {
+        return RegexDfa(
+            RegexFactory.createConcat(
+                RegexFactory.createAtomic(setOf('a')),
+                RegexFactory.createAtomic(setOf('b'))
+            )
+        )
+    }
+
+    @Test fun `walk on empty regex is dead`() {
```

Comment:
Actually, I used this naming style as well. Adding `test` before the name seems a bit redundant since there is `@Test` annotation. We may discuss this later and then refactor the code to make it consistent.